File: mycuration/src/main/java/com/phicdy/mycuration/presentation/view/ArticleSearchResultView.java
package com.phicdy.mycuration.presentation.view;

import android.support.annotation.NonNull;

import com.phicdy.mycuration.rss.Article;

import java.util.ArrayList;

public interface ArticleSearchResultView {
    void refreshList(@NonNull ArrayList<Article> articles);
    void startInternalWebView(@NonNull String url);
    void startExternalWebView(@NonNull String url);
    void startShareUrl(@NonNull String url);
}

File: mycuration/src/main/java/com/phicdy/mycuration/presentation/view/ArticleSearchResultView.kt
package com.phicdy.mycuration.presentation.view

import com.phicdy.mycuration.data.rss.Article

import java.util.ArrayList

interface ArticleSearchResultView {
    fun refreshList(articles: ArrayList<Article>)
    fun startInternalWebView(url: String)
    fun startExternalWebView(url: String)
    fun startShareUrl(url: String)
}
